### ISSUE TITLE
When searching for a word which doesn't exist nothing happens

### ISSUE THREAD

**Issue by clapierre** [benetech/Imageshare-Extensions#30]
chrome: Canary mv3 (Mac)

When searching for a word which doesn't exist or multiple words which also result in nothing found, the user is not informed of 0 results found.

IE: search for "wikipedia" or "color shape depth" no feedback given.

**Comment by clapierre** [comment]
Here is the results from the inspector:

background.js:1 Service-worker has loaded via background.js.
background.js:186 Selection Object: "Wikipedia"
background.js:31 No results found for Wikipedia
background.js:40 response null
background.js:186 Selection Object: "Wikipedia"
background.js:31 No results found for Wikipedia
Error handling response: TypeError: Cannot read property 'id' of undefined
    at chrome-extension://ppjnmachmnhefkmcnnmmjmcecjlhjlof/background.js:34:43

**Comment by LE0616** [comment]
@jkva I'm having trouble duplicating this error on my end. All the "no results found" notifications are working for me. How about on your end?

**Comment by sinabahram** [comment]
@LE0616  this line:
Error handling response: TypeError: Cannot read property 'id' of undefined

Tells me that a different path to the above is getting executed or that perhaps after the first "no results" path, the second time is stateful in a way we don't want. But, those two places are where I would start to debug.

**Comment by jkva** [comment]
This seems related to #33.

**Comment by clapierre** [comment]
I am still seeing the behavior where I am not getting any notifications.

**Comment by jkva** [comment]
Should be addressed as of 41f183d15b9d583f79caac0f358d6659bd402f2c. @clapierre can you verify?

**Comment by clapierre** [comment]
Still nothing happens on my system Mac with Chome Dev/Canary and FF, nothing happens with either right click or from the menubar search button.

**Comment by jkva** [comment]
I'm not able to reproduce this on Chrome mv2, mv3 and firefox (haven't tried safari yet) - looks like today's changes have addressed this. Leaving open for now in case @clapierre can confirm.

**Comment by sinabahram** [comment]
@clapierre  please reopen if you can reproduce with exact steps, screenshots, or anything else.

**Comment by clapierre** [comment]
I don't know what to say but for me I just tried to Chrome Canary I just selected some random text on a page "never had such functions", right clicked and said search Imageshare.  Nothing happened.

When I went to the Plugin options page I did see this which may be the problem?

Unchecked runtime.lastError: Could not establish connection. Receiving end does not exist.
Context
popup.html
Stack Trace
popup.html:0 (anonymous function)
Nothing to see here, move along.

Happy to do a zoom session with someone to try to work through this.  I have never seen this work BTW.
Maybe a Mac issue with permissions to bring up a notification window when something isn't found.

**Comment by clapierre** [comment]
Update.
Ok in Chrome Dev, It finally brought up a question if I allow for this website to send me a notification.  When I accepted that I do see the notification in the top right corner of the page.  Hard to see, would be better if it was centered on the screen but thats minor.  However in Chrome Canary I didn't get this same notification and when searching for something which doesn't exist nothing happens.  Is there some setting I need to turn on manually?  Do we need to document that?

**Comment by clapierre** [comment]
With Firefox, even with "Block pop-up windows" disabled I don't get anything showing up.

**Comment by jkva** [comment]
@clapierre which os and Firefox version is that?

This screenshot is OSX Catalina, Firefox 89

![A screenshot of slashdot open in a firefox browser, firefox shows a notification about imageshare search results.](https://user-images.githubusercontent.com/217264/120930305-ffbba080-c6ec-11eb-9de9-3570110e2dfc.png)

**Comment by clapierre** [comment]
Mac OS 10.15.7 (19H524)
FireFox Developers Edition 90.0b4 (64 bit)

Just tried again very odd no popups and I turned on voice over and didn't hear anything either.
Very strange.
@jkva I can make a video if you would like showing this.

**Comment by jkva** [comment]
@clapierre I believe you :) will try this out on the developer's edition as well.

**Comment by jkva** [comment]
Works fine for me in the developer edition as well 😕 I'm not sure why I can't reproduce the problem. It works with the development build as well as the production build.

![A screenshot showing a notification popping up after a standard search is performed.](https://user-images.githubusercontent.com/217264/121150224-9ef2ac00-c843-11eb-92af-5b054286555c.png)

**Comment by clapierre** [comment]
I see the # of results change from say 20 from my previous search to 0 but nothing else happens.
Both Firefox Dev, and Google Chrome Canary.  
Wonder if every time I install the plugin I need to adjust some settings to allow popups from the extension?

**Comment by jkva** [comment]
@clapierre not sure -- I notice the current MV3 build crashes in Canary but works fine in Chrome stable. Can you make me a recording of it not working in Firefox Dev?

**Comment by clapierre** [comment]
@jkva here is the [FF Dev video](https://benetech.box.com/s/z5qbgtnpai0n3htm63hnygkahmxagea7) showing that I am not getting any notification that 0 results were found.
also as you see in the video advance search doesn't seem to change the cursor to wait cursor while processing.

**Comment by jkva** [comment]
Do any other websites manage to show browser notifications? Are they possibly turned off somewhere?
Everything in that video otherwise seems correct. I've tested locally with the same build of the extension, same browser, same wiki page -- no problems. All I can think of is a notification setting. Do you see any errors or otherwise indicators in the inspection of the background script?

**Comment by clapierre** [comment]
Ok, I finally figured this out, there is an OS level Notifications settings for each of the browsers and instances of browsers that I have installed and it defaults to turning off notifications.

Sorry for all this, but I see now why I got very inconsistent results because for some reason one of the instances had notifications on, but the rest (Chrome Canary, FF Dev. etc.) had them off.

Change preferences for notifications

Change how website notifications appear: Choose Apple menu  > System Preferences, click Notifications, select the browser on the left, then change the notification alert style and other settings on the right.


I did find one small issue where the text from the notification was cut off but I will make that a separate issue.  This one now is closed.
